pkg/controllers/metrics/state/node.go

```diff
@@ -0,0 +1,238 @@
+limitations under the License.
+*/
+
+package metrics
+
+import (
+	"context"
+	"strings"
+
+	"github.com/aws/karpenter/pkg/apis/provisioning/v1alpha5"
+	"github.com/aws/karpenter/pkg/controllers/state"
+	"github.com/aws/karpenter/pkg/utils/resources"
+	"github.com/prometheus/client_golang/prometheus"
+	v1 "k8s.io/api/core/v1"
+	"knative.dev/pkg/logging"
+	crmetrics "sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+const (
+	resourceType    = "resource_type"
+	nodeName        = "node_name"
+	nodeProvisioner = "provisioner"
+	nodePhase       = "phase"
+)
+
+var (
+	allocatableGaugeVec    = newNodeGaugeVec("allocatable", "Node allocatable are the resources allocatable by nodes. Labeled by provisioner name, node name, zone, architecture, capacity type, instance type, node phase and resource type.")
+	podRequestsGaugeVec    = newNodeGaugeVec("total_pod_requests", "Node total pod requests are the resources requested by non-DaemonSet pods bound to nodes.  Labeled by provisioner name, node name, zone, architecture, capacity type, instance type, node phase and resource type.")
+	podLimitsGaugeVec      = newNodeGaugeVec("total_pod_limits", "Node total pod limits are the resources specified by non-DaemonSet pod limits. Labeled by provisioner name, node name, zone, architecture, capacity type, instance type, node phase and resource type.")
+	daemonRequestsGaugeVec = newNodeGaugeVec("total_daemon_requests", "Node total daemon requests are the resource requested by DaemonSet pods bound to nodes. Labeled by provisioner name, node name, zone, architecture, capacity type, instance type, node phase and resource type.")
+	daemonLimitsGaugeVec   = newNodeGaugeVec("total_daemon_limits", "Node total pod limits are the resources specified by DaemonSet pod limits. Labeled by provisioner name, node name, zone, architecture, capacity type, instance type, node phase and resource type.")
+	overheadGaugeVec       = newNodeGaugeVec("system_overhead", "Node system daemon overhead are the resources reserved for system overhead, the difference between the node's capacity and allocatable values are reported by the status. Labeled by provisioner name, node name, zone, architecture, capacity type, instance type, node phase and resource type.")
+	wellKnownLabels        = getWellKnownLabels()
+)
+
+func newNodeGaugeVec(name, help string) *prometheus.GaugeVec {
+	return prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: "karpenter",
+			Subsystem: "nodes",
+			Name:      name,
+			Help:      help,
+		},
+		nodeLabelNames(),
+	)
+}
+
+func nodeLabelNames() []string {
+	labels := []string{
+		resourceType,
+		nodeName,
+		nodeProvisioner,
+		nodePhase, // NOTE: deprecated
+	}
+
+	for _, l := range wellKnownLabels {
+		labels = append(labels, l)
+	}
+
+	return labels
+}
+
+type nodeScraper struct {
+	cluster  *state.Cluster
+	labelMap map[string]map[*prometheus.GaugeVec][]prometheus.Labels
+}
+
+func newNodeCollector(cluster *state.Cluster) *nodeScraper {
```

Comment:
Is this a collector or a scraper?

Reply:
Scraper, I must have missed that when renaming